### PR TITLE
fix: trigger slow sync when it is needed after app update [AR-3424]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
After adding some feature flags from AR kalium doesn't know to trigger slow sync

### Causes (Optional)
Some features can be disabled until app will perform slow sync (max 7 days delay)

### Solutions
Add manual trigger to perform slow sync when there is need.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManagerTest.kt [hunk 1]
```diff
@@ -412,6 +465,20 @@ class SlowSyncManagerTest {
...
+                .whenInvoked()
+                .thenReturn(0)
+        }
+
+        fun withNewLastSlowSyncVersion() = apply {
+            given(slowSyncRepository)
+                .suspendFunction(slowSyncRepository::getSlowSyncVersion)
+                .whenInvoked()
+                .thenReturn(Int.MAX_VALUE)
+        }
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManagerTest.kt:480]: Just the naming of these functions are ambiguous. They are talking about the last slow sync performed, right? Not the CURRENT Slow Sync version.

```suggestion
        fun withLastSlowSyncPerformedOnAnOldVersion() = apply {
            given(slowSyncRepository)
                .suspendFunction(slowSyncRepository::getSlowSyncVersion)
                .whenInvoked()
                .thenReturn(0)
        }

        fun withLastSlowSyncPerformedOnANewVersion() = apply {
            given(slowSyncRepository)
                .suspendFunction(slowSyncRepository::getSlowSyncVersion)
                .whenInvoked()
                .thenReturn(Int.MAX_VALUE)
        }
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt [hunk 2]
```diff
@@ -152,5 +155,6 @@ internal class SlowSyncManager(
         val MIN_RETRY_DELAY = 1.seconds
         val MAX_RETRY_DELAY = 10.minutes
         val MIN_TIME_BETWEEN_SLOW_SYNCS = 7.days
+        const val CURRENT_VERSION = 3
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt:158]: Because `1` was too mainstream? 😆

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Looks good to me 👨‍🍳🤌🏻

Just a small naming clarity suggestion.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt [hunk 3]
```diff
@@ -103,7 +103,9 @@ internal class SlowSyncManager(
                 logger.i("Last SlowSync was performed on '$lastTimeSlowSyncWasPerformed'")
                 val nextSlowSyncDateTime = lastTimeSlowSyncWasPerformed + MIN_TIME_BETWEEN_SLOW_SYNCS
                 logger.i("Next SlowSync should be performed on '$nextSlowSyncDateTime'")
-                currentTime > nextSlowSyncDateTime
+                val lastVersion = slowSyncRepository.getSlowSyncVersion()
+                logger.i("Last saved SlowSync version is $lastVersion, current is $CURRENT_VERSION")
+                currentTime > nextSlowSyncDateTime || CURRENT_VERSION > lastVersion
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt:108]: 👏

Comment (gongracr) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt:108]: We should add some documentation though, as to when is it advisable to bump the `CURRENT_VERSION`. Maybe also add it to the `FeatureConfigEventReceiver` and/or the `SyncFeatureConfigsUseCase`

**Review by gongracr — APPROVED** [review]
Looking good. I would still try to add some documentation on why we need to bump the `CURRENT_VERSION` or maybe show some examples. Graceful way to implement this in any case 🦋 🙇

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt (hunk 2 above)
Comment (vitorhugods) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt:158]: All my friends start counting at 3

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌🏻

**Pull request merged**